inject/src/main/java/io/micronaut/context/event/BeanCreatedEvent.java

```diff
@@ -48,4 +64,11 @@ public BeanCreatedEvent(BeanContext beanContext, BeanDefinition<T> beanDefinitio
     public BeanIdentifier getBeanIdentifier() {
         return beanIdentifier;
     }
+
+    /**
+     * @return The argument used to create the bean
+     */
+    public Argument<T> getBeanType() {
```

Comment:
please add `@since 4.9.0` and `@io.micronaut.core.annotation.NonNull` to the return type

Reply:
Done